src/foam/nanos/analytics/AnalyticEvent.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.analytics',
+  name: 'AnalyticEvent',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'id'
+    },
+    {
+      class: 'String',
+      name: 'name'
+    },
+    { 
+      class: 'DateTime',     
+      name: 'timestamp',
+    },
+    {
+      class: 'Int',
+      name: 'duration',
+      units: 's'
+    },
+    {
+      class: 'StringArray',
+      name: 'tags'
+    },
+    {
+      class: 'String',
+      name: 'traceId'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.session.Session',
+      name: 'sessionId'
+    },
+    {
+      class: 'Object',
```

Comment:
Should this be a reference?

Reply:
A reference to what?

Reply:
@sarthak-marwaha reference to what?  And we can't have one property be a Reference to multiple other types, so no, a reference would not work. 
This is the objects id, and if the id is generated with FUID then we can use FUID to figure out what DAO it came from.

Reply:
Right....references need a single DAO, maybe FObjectProperty then